addons/website_sale/static/src/js/website_sale.editor.js

```diff
@@ -160,4 +176,64 @@ options.registry.website_sale = options.Class.extend({
         }).then(this.reload);
     }
 });
+
+/**
+ * Handles the edition of snippet's products search bar.
+ */
+options.registry.ProductsSearchBar = options.Class.extend({
+    xmlDependencies: ['/website_sale/static/src/xml/website_sale.editor.xml'],
+
+    /**
+     * @override
+     */
+    start: function () {
+        this.$searchProductsBar = this.$('.o_wsale_products_search_bar_input');
+        this.$searchProductsInput = this.$('.o_wsale_search_products');
+        this.$searchOrderField = this.$('.o_wsale_search_order_by');
+        return this._super.apply(this, arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * @see this.selectClass for parameters
+     */
+    openSearchbarSettings: function (previewMode, value, $opt) {
+        var self = this;
+        new Dialog(this, {
+            title: _t("Products Search Bar"),
+            $content: $(qweb.render('website_sale.dialog.productsSearchBar', {
+                currentDisplayDescription: this.$searchProductsInput.attr('data-display-description') === 'true',
+                currentDisplayPrice: this.$searchProductsInput.attr('data-display-price') === 'true',
+                currentDisplayImage: this.$searchProductsInput.attr('data-display-image') === 'true',
+                currentOrderBy: this.$searchOrderField.val(),
+            })),
+            buttons: [
+                {
+                    text: _t("Save"),
+                    classes: 'btn-primary',
+                    click: function () {
+                        var displayPriceChecked = this.$('#display_price').is(':checked');
+                        self.$searchProductsInput.attr({
+                            'data-display-description': this.$('#display_description').is(':checked'),
+                            'data-display-price': displayPriceChecked,
+                            'data-display-image': this.$('#display_image').is(':checked'),
+                        });
+                        self.$searchOrderField.attr({
+                            'value': this.$('.o_order_by').children('option:selected').val(),
```

Comment:
Is `.children('option:selected')` necessary ? I think `.val()` on the select element is enough ?